lib/Mojo/mysql/Migrations.pm

```diff
@@ -0,0 +1,231 @@
+package Mojo::mysql::Migrations;
+use Mojo::Base -base;
+
+use Carp 'croak';
+use Mojo::Loader;
+use Mojo::Util 'slurp';
+
+use constant DEBUG => $ENV{MOJO_MIGRATIONS_DEBUG} || 0;
+
+has name => 'migrations';
+has 'mysql';
+
+sub active { $_[0]->_active($_[0]->mysql->db) }
+
+sub from_data {
+  my ($self, $class, $name) = @_;
+  return $self->from_string(
+    Mojo::Loader->new->data($class //= caller, $name // $self->name));
+}
+
+sub from_file { shift->from_string(slurp pop) }
+
+sub from_string {
+  my ($self, $sql) = @_;
+  my ($version, $way);
+  my $migrations = $self->{migrations} = {up => {}, down => {}};
+  for my $line (split "\n", $sql // '') {
+    ($version, $way) = ($1, lc $2) if $line =~ /^\s*--\s*(\d+)\s*(up|down)/i;
+    $migrations->{$way}{$version} .= "$line\n" if $version;
+  }
+
+  return $self;
+}
+
+sub latest { (sort keys %{shift->{migrations}{up}})[-1] || 0 }
+
+sub migrate {
+  my ($self, $target) = @_;
+  $target //= $self->latest;
+
+  # Unknown version
+  my ($up, $down) = @{$self->{migrations}}{qw(up down)};
+  croak "Version $target has no migration" if $target != 0 && !$up->{$target};
+
+  # Already the right version (make sure migrations table exists)
+  my $db = $self->mysql->db;
+  return $self if $self->_active($db) == $target;
+
+  # Lock migrations table and check version again
```

Comment:
The table lock on mojo_migrations was removed. I wasn't able to make it function like the Postgres module. 49-53 checks the current version and exits the migrations if the database is on the correct version.